asv_bench/benchmarks/pandas/join_merge.py

```diff
@@ -0,0 +1,261 @@
+                "key2": np.tile(np.arange(250).repeat(10), 4),
+                "value": np.random.randn(10000),
+            }
+        )
+        self.df2 = DataFrame({"key1": np.arange(500), "value2": np.random.randn(500)})
+        self.df3 = self.df[:5000]
+
+    def time_merge_2intkey(self, sort):
+        merge(self.left, self.right, sort=sort)
+
+    def time_merge_dataframe_integer_2key(self, sort):
+        merge(self.df, self.df3, sort=sort)
+
+    def time_merge_dataframe_integer_key(self, sort):
+        merge(self.df, self.df2, on="key1", sort=sort)
+
+    def time_merge_dataframe_empty_right(self, sort):
+        merge(self.left, self.right.iloc[:0], sort=sort)
+
+    def time_merge_dataframe_empty_left(self, sort):
+        merge(self.left.iloc[:0], self.right, sort=sort)
+
+    def time_merge_dataframes_cross(self, sort):
+        merge(self.left.loc[:2000], self.right.loc[:2000], how="cross", sort=sort)
+
+
+class I8Merge:
+
+    params = ["inner", "outer", "left", "right"]
+    param_names = ["how"]
+
+    def setup(self, how):
+        low, high, n = -1000, 1000, 10**6
+        self.left = DataFrame(
+            np.random.randint(low, high, (n, 7)), columns=list("ABCDEFG")
+        )
+        self.left["left"] = self.left.sum(axis=1)
+        self.right = self.left.sample(frac=1).rename({"left": "right"}, axis=1)
+        self.right = self.right.reset_index(drop=True)
+        self.right["right"] *= -1
+
+    def time_i8merge(self, how):
+        merge(self.left, self.right, how=how)
+
+
+class MergeCategoricals:
+    def setup(self):
+        self.left_object = DataFrame(
+            {
+                "X": np.random.choice(range(0, 10), size=(10000,)),
+                "Y": np.random.choice(["one", "two", "three"], size=(10000,)),
+            }
+        )
+
+        self.right_object = DataFrame(
+            {
+                "X": np.random.choice(range(0, 10), size=(10000,)),
+                "Z": np.random.choice(["jjj", "kkk", "sss"], size=(10000,)),
+            }
+        )
+
+        self.left_cat = self.left_object.assign(
+            Y=self.left_object["Y"].astype("category")
+        )
+        self.right_cat = self.right_object.assign(
+            Z=self.right_object["Z"].astype("category")
+        )
+
+        self.left_cat_col = self.left_object.astype({"X": "category"})
+        self.right_cat_col = self.right_object.astype({"X": "category"})
+
+        self.left_cat_idx = self.left_cat_col.set_index("X")
+        self.right_cat_idx = self.right_cat_col.set_index("X")
+
+    def time_merge_object(self):
+        merge(self.left_object, self.right_object, on="X")
+
+    def time_merge_cat(self):
+        merge(self.left_cat, self.right_cat, on="X")
+
+    def time_merge_on_cat_col(self):
+        merge(self.left_cat_col, self.right_cat_col, on="X")
+
+    def time_merge_on_cat_idx(self):
+        merge(self.left_cat_idx, self.right_cat_idx, on="X")
+
+
+class Align:
```

Comment:
I'm wondering how important is this in terms of modin execution paths? We don't want to run the same functionality several times